doc/docs.md

```diff
@@ -847,13 +948,78 @@ which module is being called. Especially in large code bases.
 
 Module names should be short, under 10 characters. Circular imports are not allowed.
 
-You can create modules anywhere.
+Module names should be snake_case identifier (`/[a-z][a-z0-9_-]/`) as a convention. You cannot use space characters for module names.
+
+The imported modules are all automatically compiled, which means you don't need to precompile modules unless you need `.o` file.
+
+TBD: You can create modules anywhere.
+
+The lookup for modules happens in this order:
+
+1. project's local folders (like `./my_module/`)
+2. v's `vlib/` folder (like `vlib/my_module/`)
+3. home directory's `~/.vmodules/` folder (like `~/.vmodules/my_module/`)
+
+You can specify modules in subdirectory with `.` (not with `/`) like`module_name.submodule_name`:
+
+```v
+./
+  |- parent/
+    |- child/
+      |- child1.v
+  |- main.v
+
+// child1.v
+module child
+
+// main.v
+import parent.child     // imports modules under parent/child/
+
+child.function()        // reference is not namespaced
+```
+
+Notes on submodules:
+
+* The maximum depth of submodule path is "5" and you cannot exceed this.
+* TBD: When importing modules, you cannot use `../` to refer to the modules in parent directories. Use symbolic links in the case.
+
+You can rename the modules when imported:
+
+```v
+import module_name as mn      // mn is an alias of the module_name
+```
+
+Note that V looks up the module by the containing directory name, which means the directory name should match exactly the module name.
+
+The name of the modules under the directory can be any `*.v`, but at least putting the file with the exact module name is preferable as a convention because the module file with the exact name, where `init()` function (see below) can exist, will be fetched first.
```

Comment:
> where `init()` function (see below) can exist, will be fetched first.
That is not so.
The filenames in a module just need to end with `.v` . The init() function for a module can be anywhere inside one of the `.v` files.

Reply:
Thank you! Fixed as suggested.